include/session/onionreq/builder.hpp

```diff
@@ -1,12 +1,42 @@
 #pragma once
 
+#include <oxen/quic/address.hpp>
 #include <string>
 #include <string_view>
+#include <variant>
 
 #include "key_types.hpp"
 
 namespace session::onionreq {
 
+struct ServerDestination {
+    std::string protocol;
+    std::string host;
+    std::string endpoint;
+    session::onionreq::x25519_pubkey x25519_pubkey;
+    std::optional<uint16_t> port;
+    std::optional<std::vector<std::pair<std::string, std::string>>> headers;
+    std::string method;
+
+    ServerDestination(
+            std::string protocol,
+            std::string host,
+            std::string endpoint,
+            session::onionreq::x25519_pubkey x25519_pubkey,
+            std::optional<uint16_t> port,
+            std::optional<std::vector<std::pair<std::string, std::string>>> headers,
```

Comment:
Perhaps these two should get ` = std::nullopt` defaults?

Reply:
These defaults have now been added